geocropper/utils.py

```diff
@@ -2123,3 +2124,74 @@ def standardize_rasterio_image(rasterio_image, standardization_procedure="layerw
         standardized_image_array = numpy.reshape(standardized_image_array, newshape=(num_layers, num_pixel_y, num_pixel_x))
     
     return standardized_image_array
+
+
+def reduce_image_dimensionality(image_dir, image_name, target_dir, crop, dim_reduction_method):
+    """Reduces the dimension of a stacked image by extracting the most relevant information of each layer:
+    
+    Parameters
+    ----------
+    image_dir: Path
+        Path of sentinel 1 standardized images (VV and VH)
+    image_name: String
+        Name of the image which dimension shall be reduced ("s1_standardized_VV.tif" or "s1_standardized_VH.tif")
+    target_dir: Path
+        Default is source_dir with prefix "dim_reduced_"
+    crop: String
+        Location of the crop in latitude and longitude format as a string (e.g. "-68.148781_44.77383")
+    dim_reduction_method: String, optional
+        Set the method for dimension reduction (default is "pca")
+        "pca" = Principal Component Analysis, simplify data with a small amount of linear components
+        "max_values" = The maximum value of each pixel from all layers is selected
+    """
+

```

Comment:
Check if module otbApplication is loaded, if not => return 
only for pca dim_reduction_method